data-structures/src/test/java/com/baeldung/circularbuffer/ProducerConsumerLiveTest.java

```diff
@@ -0,0 +1,77 @@
+package com.baeldung.circularbuffer;
+
+import static org.junit.Assert.assertArrayEquals;
+
+import java.util.concurrent.locks.LockSupport;
+
+import org.junit.jupiter.api.Test;
+
+public class ProducerConsumerLiveTest {
+
+    private final String[] shapes = { "Circle", "Triangle", "Rectangle", "Square", "Rhombus", "Trapezoid", "Pentagon", "Pentagram", "Hexagon", "Hexagram" };
+
+    @Test
+    public void givenACircularBuffer_whenInterleavingProducerConsumer_thenElementsMatch() throws InterruptedException {
+        CircularBuffer<String> buffer = new CircularBuffer<String>(shapes.length);
+        String[] consumedShapes = new String[shapes.length];
+
+        Thread producer = new Thread(new Producer(shapes, buffer));
```

Comment:
let's use the concurrent api (Executors...)

Reply:
changed